json-schema-processor/src/main/java/io/micronaut/jsonschema/visitor/aggregator/JacksonInfoAggregator.java

```diff
@@ -0,0 +1,160 @@
+                    }
+                }
+
+                element.stringValue(JsonPropertyDescription.class)
+                    .ifPresent(schema::setDescription);
+                if (property.hasAnnotation(JsonUnwrapped.class)) {
+                    schema.getProperties().remove(property.getName());
+                    schema.getProperties().putAll(propertySchema.getProperties());
+                } else {
+                    property.stringValue(JsonProperty.class).ifPresent(newName -> {
+                        schema.getProperties().remove(property.getName());
+                        schema.putProperty(newName, propertySchema);
+                    });
+                }
+            }
+        }
+    }
+
+    private void addSubtypeInfo(ClassElement element, Schema schema, VisitorContext visitorContext, JsonSchemaContext context) {
+        AnnotationValue<?> subTypesAnn = element.getAnnotation(JsonSubTypes.class);
+        AnnotationValue<?> typeInfoAnn = element.getAnnotation(JsonTypeInfo.class);
+        if (subTypesAnn == null || typeInfoAnn == null) {
+            return;
+        }
+        JsonTypeInfo.Id id = typeInfoAnn.enumValue("use", JsonTypeInfo.Id.class).orElse(Id.NAME);
+        String discriminatorName = typeInfoAnn.stringValue("property")
+            .orElse(id.getDefaultPropertyName());
+
+        for (AnnotationValue<?> subTypeAnn: subTypesAnn.getAnnotations("value", JsonSubTypes.Type.class)) {
+            ClassElement subType = subTypeAnn.stringValue()
+                .flatMap(visitorContext::getClassElement).orElse(null);
+            if (subType != null) {
+                Schema subTypeSchema = JsonSchemaVisitor.createSchema(subType, visitorContext, context);
+                schema.addOneOf(subTypeSchema);
+
+                if (discriminatorName != null) {
+                    String discriminatorValue = null;
+                    if (id == Id.MINIMAL_CLASS) {
+                        discriminatorValue = getMinimalClassName(element.getPackageName(), subType.getName());
+                    } else if (id == Id.NAME) {
+                        if (subTypeAnn.stringValues("names").length != 0) {
+                            subTypeSchema.putProperty(discriminatorName, Schema.string().setEnumValues(
+                                Arrays.stream(subTypeAnn.stringValues("names")).map(v -> (Object) v).toList()
+                            ));
+                        } else {
+                            discriminatorValue = subTypeAnn.stringValue("name")
+                                .orElse(subType.stringValue(JsonTypeName.class).orElse(subType.getSimpleName()));
+                        }
+                    } else {
+                        discriminatorValue = subType.getName();
+                    }
+
+                    if (discriminatorValue != null) {
+                        subTypeSchema.putProperty(discriminatorName, Schema.string().setConstValue(discriminatorValue));
```

Comment:
what about other `include` settings

Reply:
Oh no. The [EXTERNAL_PROPERTY](https://fasterxml.github.io/jackson-annotations/javadoc/2.4/com/fasterxml/jackson/annotation/JsonTypeInfo.As.html#EXTERNAL_PROPERTY) one is just horrible.

Do we support all of them in serde?

Reply:
I think we support these: https://github.com/micronaut-projects/micronaut-serialization/blob/225ecba1717c27c5d7713355dbf10ec814907b2c/serde-api/src/main/java/io/micronaut/serde/config/annotation/SerdeConfig.java#L345